doc/setup_instructions.md

```diff
@@ -486,3 +485,117 @@ build tool option, if it isn't already.
+Just like Python, you'll want to download these if you are planning
+to add or edit strings into the game.
+
+### Optional : Poedit
+
+[Poedit](https://poedit.net/) is a free .pot and .po file editor that may
+make your life easier when working on translation files.
+
+It is needed to create new .po files without using the command line tools.
+
+Note: Poedit will complain about translation format since it was made to
+directly use text as keys. Those can be ignored.
+
+
+Adding new contents into the game with translation in mind
+----------------------------------------------------------
+
+Working with translation in mind will be a bit different than usual, as
+it will require a few more steps when working with strings.
+
+### Working in scene files
+
+When working on a scene, once you are done designing it, take note
+of all the strings (text, titles, ...) somewhere.
+
+Replace all the strings in your scene with keys (eg. AUTOEVO_POPULATION_CHANGED) and "match" them with your
+strings.
+
+(You can use a simple text file writing str => key, or anything else you prefer)
+
+### Working in cs files
+
+Always call TranslationServer.Translate() for strings that should be localized.
+
+Other than that, it is the same principle has for the scene files :
+once you are done, write down your strings somewhere And change them in the code into keys.
+
+### Updating the localizations
+
+Once you are done adding content into the game, go into the scripts folder and
+run "update_localization.rb". This will extract the strings from the game files,
+and also update the .po files if they have been added to it.
+
+The final step is to open en.po in the locale folder (you can use a text editor
+or Poedit), search for your keys, and add your strings as translation.  Once done,
+you can launch the game and make sure everything works as expected.
+
+Translating the game into a new language
+----------------------------------------
+
+### Only the first time: create your locale .po file
+
+To create a new .po file for your localization,
+you have two choices: using the commands, or Poedit.
+
+#### With commands
+
+Execute the following command in the locale folder :
+
+msginit --no-translator --input=messages.pot --locale=language_code,
+
+#### With Poedit
+
+Open Poedit and use it to generate the .po file by going into the menu
+File/New from POT/PO file...
+
+**In both cases, you'll need to enter your language code. You can find a list of all
+supported code [in the Godot engine documentation](https://docs.godotengine.org/en/stable/tutorials/i18n/locales.html)**
+
+#### Add your .po file to the update script
+
+To make updating the localization easier, you should add a reference to
+the new .po file into update_localization.rb.
+
+Simply open the ruby script into any text editor, and add a new line at the end
+of the .po files list as such :
+
+runOpen3Checked 'msgmerge', '--update', '--backup=none', '../locale/**new_po_file_name.po**',
```

Comment:
I think you should mention here, that copy one of the existing locale lines and modify, instead of having the template here, as this will become outdated, probably.
Also one thought regarding that, it'd be nicer to have a %w list at the top of the ruby script so that people can just write the new locale names in it instead of doing extensive copy-pasting.

```
LOCALES = %w[en fr]

# and then later
LOCALES.each{|locale|
do the script running...
}
```